docs/developer-docs/setup/playground.md

```diff
@@ -0,0 +1,279 @@
+
+Each canister has an initial amount of 0.5T cycles and can be used for 20 minutes once the canister has been deployed. To avoid wasting cycles, canisters cannot transfer cycles, and the cycles transfer instructions have been removed.
+
+To ensure resources are used fairly by users, proof of work when a user makes a request for a canister ID is required. 
+
+Since Motoko playground is designed for short, small-scale testing, there are limitations and restrictions imposed on canisters deployed to the playground. These limitations are:
+
+- Cycle transfer instructions are silently ignored by the system.
+- Canisters can use at most 1GB of stable memory.
+- Canisters can call the management canister to manage itself without being the controller.
+- Deployed canisters expire after 20 minutes. Upgrades to canisters reset this timer. When the timer runs out, the canister(s) will be uninstalled. 
+
+## `dfx deploy --playground`
+
+Through `dfx`, canisters can be deployed directly from the CLI to the Motoko playground using the `--playground` flag. When this flag is used, the canisters in the `dfx` project will be deployed to the IC mainnet using borrowed resources from the Motoko playground canister pool, and a canister URL will be returned to the developer. This enables developers to test canister functionality or workflow without needing to set up a cycles wallet, obtain cycles, or configure other parameters that are required for deployment to the mainnet. 
+
+### Prerequisites
+
+Before you start, verify the following:
+
+- [x] You have an internet connection and access to a shell terminal on your local macOS or Linux computer.
+
+- [x] You have a command line interface (CLI) window open. This window is also referred to as the 'terminal' window.
+
+- [x] You have downloaded and installed the IC SDK package as described in the [installing the IC SDK](./install/index.mdx) page.
+
+
+:::info
+`dfx deploy --playground` is available in `dfx` versions v0.15.1 and above.
+:::
+
+To deploy a canister to the Motoko playground, first create a new `dfx` project with the command:
+
+```
+dfx new hello_world
+```
+
+This command will create a new default `dfx` project, which contains a simple 'Hello, world!' sample.
+
+Then, to deploy this canister to the playground, run the command:
+
+```
+dfx deploy --playground
+```
+
+Once deployed, the canister can be interacted with using a command such as:
+
+```
+dfx canister --network playground call hello__world_backend greet '("everyone": text)'
+```
+
+This command calls the `hello_world_backend` canister that has been deployed to the playground using the `--network` flag, since the playground is classified as a deployment network. 
+
+:::info
+Any commands that intend to target a canister deployed to the playground must use the `--playground` or `--network playground` flag in order to target the borrowed canister(s). 
+:::
+
+### Defining custom playground networks
```

Comment:
I think this section should be expanded:

What can custom playgrounds be used for?
- Sharing a canister pool for a team so they don't have to manage cycles wallets for development
- Using a canister pool for CI (preview?) deployments without having to create/delete canisters all the time

How to do it:
- clone https://github.com/dfinity/motoko-playground
- Add access control (if desired)
- Change `install_code` to skip WASM analysis if desired
  - also allows much larger wasms to be installed because analysis takes a _lot_ of instructions
- deploy canisters `backend` and, if desired `wasm-utils`
- configure network as defined below
- If the network shall be the default so that one doesn't have to type `--playground` or `--network custom-playground` the `local` network can be defined, which is the network that's always implicitly picked